importer_test.go

```diff
@@ -43,193 +40,165 @@ func TestAppendSearchFromEnvEmpty(t *testing.T) {
-
-func TestImport(t *testing.T) {
-	f := &fetcherMock{statusCode: http.StatusOK, content: "hello world"}
-	i := Importer{Fetcher: f}
-
-	contents, location, err := i.Import("/a/y", "x")
-
+func TestImportLocal(t *testing.T) {
+	i := Importer{}
+	contents, foundAt, err := i.Import("", "testdata/importer/hello.txt")
 	require.NoError(t, err)
-	require.Equal(t, "hello world", contents.String())
-	require.Equal(t, "/a/x", location)
+	require.Equal(t, "hello world\n", contents.String())
+	require.Equal(t, "testdata/importer/hello.txt", foundAt)
 }
 
-func TestImportNotFound(t *testing.T) {
-	f := &fetcherMock{statusCode: http.StatusNotFound, content: ""}
-	i := Importer{Fetcher: f}
-
-	_, _, err := i.Import("/a/y", "x")
-
+func TestImportLocalNotFound(t *testing.T) {
+	i := Importer{}
+	_, _, err := i.Import("", "testdata/importer/notfound.txt")
 	require.Error(t, err)
 }
 
-func TestImportNetpathSource(t *testing.T) {
-	f := &fetcherMock{statusCode: http.StatusNotFound, content: ""}
-	i := Importer{Fetcher: f}
-
-	_, _, err := i.Import("//example.com/y", "x")
-
+// Make reading file by trying to read a directory. That causes read(2) to
+// return EDIR.
+func TestImportLocalReadError(t *testing.T) {
+	i := Importer{}
+	_, _, err := i.Import("", "testdata/importer")
 	require.Error(t, err)
-	require.Equal(t, []string{"https://example.com/x"}, f.urls)
 }
 
-func TestImportRootNetpathSource(t *testing.T) {
-	f := &fetcherMock{statusCode: http.StatusNotFound, content: ""}
-	i := Importer{Fetcher: f}
-
-	// Should try to import `//example.com/x`, not `//x`
-	_, _, err := i.Import("//example.com", "x")
-
-	require.Error(t, err)
-	require.Equal(t, []string{"https://example.com/x"}, f.urls)
+func TestImportLocalSourceRelative(t *testing.T) {
+	i := Importer{}
+	contents, foundAt, err := i.Import("testdata/importer/hello.txt", "mellow.txt")
+	require.NoError(t, err)
+	require.Equal(t, "mellow world\n", contents.String())
+	require.Equal(t, "testdata/importer/mellow.txt", foundAt)
 }
 
-func TestSearchAbsolute(t *testing.T) {
-	f := &fetcherMock{statusCode: http.StatusOK, content: "hello world"}
-	i := Importer{Fetcher: f}
-
-	_, location, err := i.search("/x/y/z", "/a")
-
+func TestImportLocalRelativeSearch(t *testing.T) {
+	i := Importer{}
+	i.SearchPath = []string{"testdata"}
+	contents, foundAt, err := i.Import("", "importer/hello.txt")
 	require.NoError(t, err)
-	require.Equal(t, "/x/y/z", location)
-	require.Equal(t, []string{"file:///x/y/z"}, f.urls)
+	require.Equal(t, "hello world\n", contents.String())
+	require.Equal(t, "testdata/importer/hello.txt", foundAt)
 }
 
-func TestSearchRelativeToDir(t *testing.T) {
-	f := &fetcherMock{statusCode: http.StatusNotFound, content: ""}
-	i := Importer{Fetcher: f}
-
-	_, location, err := i.search("x", "/a")
+func TestImportNetpath(t *testing.T) {
+	s := httptest.NewTLSServer(http.FileServer(http.Dir("testdata")))
+	defer s.Close()
+	np := strings.TrimPrefix(s.URL, "https:")
 
+	i := Importer{Fetcher: s.Client()}
```

Comment:
does it not work with the default client??

Reply:
as discussed, this is needed as the `httptest` server is set up with TLS and `s.Client()` is loaded with the CA cert.